### PR TITLE
Add LevelDetailsDialog and implement three level types

### PR DESCRIPTION
**Pull request by bethanyaconnor**
In an effort to avoid a 600+ line PR, I realized I could break the LevelDetails work into multiple PRs using Storybook. This means that this PR doesn't connect this dialog to the lesson plans.

This PR implements three level types:
StandaloneVideo:
Screen Shot 2021-02-16 at 2 50 32 PM

External:
Screen Shot 2021-02-16 at 2 50 42 PM

LevelGroup (text adapted from CurriculumBuilder):
Screen Shot 2021-02-16 at 4 15 35 PM

## Links

- [spec](https://docs.google.com/document/d/1sQaJzN8nctTVS2F4zQy72jzM5DNO0wiOyFJqYwdWph8/edit?usp=sharing)

## Testing story

This PR is all testing 😅 . Added both Storybook and unit tests.

## PR Checklist:

- [ ] Tests provide adequate coverage
- [ ] Privacy and Security impacts have been assessed
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

File: apps/src/templates/lessonOverview/activities/LevelDetailsDialog.jsx [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+const VIDEO_HEIGHT = 375;
+const VIDEO_MODAL_WIDTH = 700;
+
+export default class LevelDetailsDialog extends Component {
+  static propTypes = {
+    scriptLevel: PropTypes.object.isRequired,
+    handleClose: PropTypes.func.isRequired
+  };
+
+  getContentComponent = level => {
```
Comment (dmcavoy) [inline apps/src/templates/lessonOverview/activities/LevelDetailsDialog.jsx]: nit: getComponentContent

**Review by dmcavoy — APPROVED** [review]
🎉 Yay for storybook! 

This is a great start. Can't wait to see more

File: apps/test/unit/templates/lessonOverview/activities/LevelDetailsDialogTest.jsx [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+            type: 'StandaloneVideo',
+            longInstructions: 'Some things to think about.'
+          }
+        }}
+      />
+    );
+    expect(loadVideoSpy.calledOnce).to.be.true;
+    expect(wrapper.contains('Some things to think about.')).to.be.true;
+  });
+});
```
Comment (davidsbailey) [inline apps/test/unit/templates/lessonOverview/activities/LevelDetailsDialogTest.jsx:88]: test coverage looks good!

**Review by davidsbailey — APPROVED** [review]
Nice progress! I noticed that the "popover" in the spec shows the workspace header (insructions, help & tips, etc). Not sure if this is just future work, but seeing it would really help me place that the content shown here is the instructions of the level (if that is in fact what it is).

**Comment by bethanyaconnor** [comment]
@davidsbailey Most levels will have the instructions in the popover but it's been proving a bit tricky so I thought it would be easier to get some of the simpler code in earlier.

Here's a work in progress shot of what it will likely look like when everything is done!
Screen Shot 2021-02-17 at 12 01 38 PM

**Comment by dmcavoy** [comment]
> @davidsbailey Most levels will have the instructions in the popover but it's been proving a bit tricky so I thought it would be easier to get some of the simpler code in earlier.
> 
> Here's a work in progress shot of what it will likely look like when everything is done!
> Screen Shot 2021-02-17 at 12 01 38 PM

Its probably also worth pointing out that all the level types in this PR do not use a the instructions area currently on the levels themselves

**Comment by davidsbailey** [comment]
Thanks @bethanyaconnor and @dmcavoy , all makes sense!

**Pull request merged**